Use lrzhs for address validation rather than zcashd

nuttycom opened the pull request:
Running zcashd is pretty heavy-weight, and we don't actually need it here. This moves toward using a light wallet for interaction with the zcash chain.

nuttycom committed:
Use lrzhs for address validation rather than zcashd

Running zcashd is pretty heavy-weight, and we don't actually need
it here. This moves toward using a light wallet for interaction with
the zcash chain.

dchenbecker commented on cabal.project:
```diff
@@ -26,6 +26,11 @@ source-repository-package
+source-repository-package
+    type: git
+    location: https://github.com/nuttycom/lrzhs.git
+    tag: b10c8cc245f2353dfe3f5cbd6e231082a23ced7d
```
Is it possible to tag a...git tag? I know maybe you don't want HEAD, but if there were a "stable" tag would that help with ensuring up-to-date dependencies? Alternatively, are there tools that help you audit when dependencies are behind? Same goes for the tags in the flake files

nuttycom replied:
I think that it's good to use tags when you actually stabilize things, but I expect these versions (particularly this one) to be evolving pretty quickly. With respect to the `thyme` and `true-name` dependencies, those are more stable but I keep hoping that their maintainer will upstream my patches so that I can get rid of those entirely. It's not clear they've even seen them though, despite that I've made a number of attempts to contact them.

dchenbecker commented on flake.lock:
```diff
@@ -313,7 +420,7 @@
       }
     },
-    "nixpkgs_7": {
+    "nixpkgs_8": {
```
Was this generated? This seems like an undescriptive naming scheme

nuttycom replied:
Yup, this is autogenerated.

dchenbecker commented on lib/Aftok/Currency/Zcash.hs:
```diff
@@ -137,7 +144,7 @@ rpcValidateZAddr mgr cfg addr = runExceptT $ do
         Just Z.Sprout -> Left (RPCError SproutAddress)
         Just Z.Sapling -> Right (Z.Address addr)
-      else Left $ RPCError ZAddrInvalid
+      else Left (RPCError ZAddrInvalid)
```
I'm still trying to relearn Haskell. What is this change?

nuttycom replied:
Oh... this was something where I deleted the code, replaced it with something else, and then deleted that and wrote it back to its original meaning but apparently not its original state. This is a superfluous change, I can get rid of it.

dchenbecker commented on lib/Aftok/Currency/Zcash.hs:
```diff
@@ -7,6 +7,7 @@ module Aftok.Currency.Zcash
     RPCError (..),
     ZValidateAddressErr (..),
+    ZcashConfig (..),
     ZcashdConfig (..),
```
If we don't plan to run zcashd, should this dependency be removed, or is that independent of the actual service?

nuttycom replied:
Yeah, maybe it should just be gotten rid of, or moved out to a separate library.

dchenbecker reviewed (commented):
Looks good. I'm not confident enough in my Haskell or Zcash knowledge to say everything is correct, but I think I get the gist of things

nuttycom committed:
Remove zcashd rpc support.

zcashd was only being used for address parsing and validation; it's much
simpler to use Lrzhs for these things. In the future, interaction with
the chain will be done via the light client protocol.

nuttycom committed:
Reformat via ormolu.
